tests/test-security.php

```diff
@@ -118,6 +120,71 @@ public function test__lost_password_limit() {
 
 	}
 
+	public function test__wpcom_vip_track_auth_attempt__defaults() {
```

Comment:
Tip: Use `--testdox` when running tests and you'll see how a well-written test method can help better describe what it is that is actually being tested.

Adding `@covers` [annotations](https://docs.phpunit.de/en/9.6/annotations.html#covers) (needed for strict code coverage) also means the inclusion of the function under test is not needed in the test method name:

```suggestion
	/**
	 * @covers ::wpcom_vip_track_auth_attempt
	 */
	public function test_auth_attempt_tracked() {
```

Reply:
Removing the name form the test case would make it harded for developer to run all tests related that method. What I used to do and our README recommend is to do:

```
./bin/test.sh --filter test__wpcom_vip_username_is_limited
``` 

Is there a way to leverage those annotations instead?

Reply:
https://docs.phpunit.de/en/9.6/annotations.html#group is probably what you could use.

Not sure I knew about using test method prefixes to run a group of tests though! I know you can also specify a file path as well.

Reply:
It seems testsdox is happy even without the comments:

```
./bin/test.sh --filter test__wpcom_vip_username_is_limited --testdox
VIP_Go_Security_
 ✔ wpcom vip username is limited  should not limit by default
 ✔ wpcom vip username is limited  should be limit after few tries
 ✔ wpcom vip username is limited  should be limit even after the event window
 ✔ wpcom vip username is limited  should be limit after 3 attempts fedramp
 ```

Reply:
Right, but that could be arranged as something like:

```php
/**
 * @group login_limits
 * /
function test_username_should_not_be_limited_by_default() {...}
```
...which would then give an output like:
```sh
$ ./bin/test.sh --group login_limits --testdox
VIP_Go_Security
 ✔ Username should not be limited by default
 ✔ Username should be limited after few tries
 ✔ Username should be limited even after the event window
 ✔ Username should be limited after 3 attempts for Fedramp
```

You could see how the test method names can match up to acceptance criteria.

Not a blocker, and low priority, but [nicer tests output](https://peakd.com/hive-168588/@crell/advanced-phpunit-shenanigans).